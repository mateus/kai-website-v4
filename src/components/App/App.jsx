import React, { Component } from 'react';

import { ImageList, HomeHeader } from '../';
import './App.scss';

class App extends Component {
  state = {};
  mounted = true;

  componentDidMount() {
    this.loadAlbums();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  async loadAlbums() {
    const undercityAlbum = await import(/* webpackChunkName: 'undercityAlbum' */ './albums/undercity');
    const mtshastaAlbum = await import(/* webpackChunkName: 'undercityAlbum' */ './albums/mt_shasta_17');
    const canadaPNWAlbum = await import(/* webpackChunkName: 'undercityAlbum' */ './albums/canada_pnw_18');

    if (!this.mounted) {
      return;
    }

    this.setState({
      undercityAlbum,
      mtshastaAlbum,
      canadaPNWAlbum,
    });
  }

  render() {
    const { undercityAlbum, canadaPNWAlbum, mtshastaAlbum } = this.state;

    if (!undercityAlbum) {
      return <h1>Loading albums</h1>;
    } else {
      return (
        <main className="Page">
          <div className="Layout">
            <div className="LayoutItem LayoutItem--fill LayoutItem--center">
              <HomeHeader />
            </div>
            <div className="LayoutItem">
              <ImageList
                images={[
                  {
                    url: undercityAlbum.thumbnail.src,
                    title: 'Undercity ‘13 – Today',
                    description:
                      'An anthology of trips, this series is an ongoing album documenting what many would consider to be Urban Exploration',
                  },
                  {
                    url: canadaPNWAlbum.thumbnail.src,
                    title: 'Canada & PNW ‘17',
                    description:
                      'An anthology of trips, this series is an ongoing album documenting what many would consider to be Urban Exploration',
                  },
                  {
                    url: mtshastaAlbum.thumbnail.src,
                    title: 'Mt. Shasta ‘18',
                    description:
                      'An anthology of trips, this series is an ongoing album documenting what many would consider to be Urban Exploration',
                  },
                ]}
              />
            </div>
          </div>
        </main>
      );
    }
  }
}

export default App;
