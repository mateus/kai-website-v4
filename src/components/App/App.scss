@import '../../stylesheets/base';

.Page {
  display: flex;
  min-height: 100vh;
}

.Layout {
  flex: 1;
  display: flex;
  flex-direction: column;
  margin: 90px 60px;

  @include breakpoint-before(small) {
    margin: 90px spacing(loose);
  }
}

.LayoutItem {
  flex: 0 0 auto;
}

.LayoutItem--center {
  align-items: center;
}

.LayoutItem--fill {
  flex: 1;
}

.Content {
  display: flex;
  margin-top: spacing();
}

.SocialLinks {
  flex: 1;
  display: flex;
  justify-content: flex-end;
  padding-left: spacing(loose);
  line-height: rem(16px);

  li:not(:first-child) {
    list-style-type: disc;
    margin-left: spacing(loose);
  }

  @include breakpoint-before(medium) {
    position: absolute;
    top: spacing(loose);
    right: spacing(loose);
  }
}

.Title {
  font-family: 'TiemposHeadline';
}

.Quote {
  max-width: 700px;
  font-size: rem(24px);
  line-height: rem(36px);
}
