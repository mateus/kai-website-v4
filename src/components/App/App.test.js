import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
import { ImageList, HomeHeader } from '../';

describe('App', () => {
  it('renders correctly', () => {
    const app = shallow(<App />);

    expect(app).toMatchSnapshot();
  });

  it('renders HomeHeader', () => {
    const app = shallow(<App />);

    expect(app.find(HomeHeader)).toBeTruthy();
  });

  it('renders ImageList', () => {
    const app = shallow(<App />);

    expect(app.find(ImageList)).toBeTruthy();
  });
});
