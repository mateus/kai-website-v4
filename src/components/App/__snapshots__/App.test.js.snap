// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Layout"
>
        <div
                className="LayoutItem LayoutItem--fill LayoutItem--center"
        >
                <header>
                        <h1
                                className="Title"
                        >
                                <span
                                        aria-label="Wave emoji"
                                        role="img"
                                >
                                        ðŸ‘‹
                                </span>
                                I'm Kai
                        </h1>
                </header>
                <div
                        className="Content"
                >
                        <p
                                className="Quote"
                        >
                                Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                <br />
                                San Francisco, California
                        </p>
                        <ul
                                className="SocialLinks"
                        >
                                <li>
                                        <a
                                                href="https://www.instagram.com/kaigenji/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                        >
                                                @kaigenji
                                        </a>
                                </li>
                                <li>
                                        <a
                                                href="mailto:kaigenji@gmail.com"
                                        >
                                                email
                                        </a>
                                </li>
                        </ul>
                </div>
        </div>
        <div
                className="LayoutItem"
        >
                SLIDER
        </div>
</div>,
      "className": "Page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="LayoutItem LayoutItem--fill LayoutItem--center"
>
            <header>
                        <h1
                                    className="Title"
                        >
                                    <span
                                                aria-label="Wave emoji"
                                                role="img"
                                    >
                                                ðŸ‘‹
                                    </span>
                                    I'm Kai
                        </h1>
            </header>
            <div
                        className="Content"
            >
                        <p
                                    className="Quote"
                        >
                                    Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                    <br />
                                    San Francisco, California
                        </p>
                        <ul
                                    className="SocialLinks"
                        >
                                    <li>
                                                <a
                                                            href="https://www.instagram.com/kaigenji/"
                                                            rel="noopener noreferrer"
                                                            target="_blank"
                                                >
                                                            @kaigenji
                                                </a>
                                    </li>
                                    <li>
                                                <a
                                                            href="mailto:kaigenji@gmail.com"
                                                >
                                                            email
                                                </a>
                                    </li>
                        </ul>
            </div>
</div>,
          <div
            className="LayoutItem"
>
            SLIDER
</div>,
        ],
        "className": "Layout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header>
                <h1
                                className="Title"
                >
                                <span
                                                aria-label="Wave emoji"
                                                role="img"
                                >
                                                ðŸ‘‹
                                </span>
                                I'm Kai
                </h1>
</header>,
              <div
                className="Content"
>
                <p
                                className="Quote"
                >
                                Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                <br />
                                San Francisco, California
                </p>
                <ul
                                className="SocialLinks"
                >
                                <li>
                                                <a
                                                                href="https://www.instagram.com/kaigenji/"
                                                                rel="noopener noreferrer"
                                                                target="_blank"
                                                >
                                                                @kaigenji
                                                </a>
                                </li>
                                <li>
                                                <a
                                                                href="mailto:kaigenji@gmail.com"
                                                >
                                                                email
                                                </a>
                                </li>
                </ul>
</div>,
            ],
            "className": "LayoutItem LayoutItem--fill LayoutItem--center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="Title"
>
                  <span
                                    aria-label="Wave emoji"
                                    role="img"
                  >
                                    ðŸ‘‹
                  </span>
                  I'm Kai
</h1>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      aria-label="Wave emoji"
                      role="img"
>
                      ðŸ‘‹
</span>,
                    "I'm Kai",
                  ],
                  "className": "Title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "Wave emoji",
                      "children": "ðŸ‘‹",
                      "role": "img",
                    },
                    "ref": null,
                    "rendered": "ðŸ‘‹",
                    "type": "span",
                  },
                  "I'm Kai",
                ],
                "type": "h1",
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="Quote"
>
                    Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                    <br />
                    San Francisco, California
</p>,
                  <ul
                    className="SocialLinks"
>
                    <li>
                                        <a
                                                            href="https://www.instagram.com/kaigenji/"
                                                            rel="noopener noreferrer"
                                                            target="_blank"
                                        >
                                                            @kaigenji
                                        </a>
                    </li>
                    <li>
                                        <a
                                                            href="mailto:kaigenji@gmail.com"
                                        >
                                                            email
                                        </a>
                    </li>
</ul>,
                ],
                "className": "Content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in",
                      <br />,
                      "San Francisco, California",
                    ],
                    "className": "Quote",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "San Francisco, California",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li>
                        <a
                                                href="https://www.instagram.com/kaigenji/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                        >
                                                @kaigenji
                        </a>
</li>,
                      <li>
                        <a
                                                href="mailto:kaigenji@gmail.com"
                        >
                                                email
                        </a>
</li>,
                    ],
                    "className": "SocialLinks",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="https://www.instagram.com/kaigenji/"
                          rel="noopener noreferrer"
                          target="_blank"
>
                          @kaigenji
</a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "@kaigenji",
                          "href": "https://www.instagram.com/kaigenji/",
                          "rel": "noopener noreferrer",
                          "target": "_blank",
                        },
                        "ref": null,
                        "rendered": "@kaigenji",
                        "type": "a",
                      },
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="mailto:kaigenji@gmail.com"
>
                          email
</a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "email",
                          "href": "mailto:kaigenji@gmail.com",
                        },
                        "ref": null,
                        "rendered": "email",
                        "type": "a",
                      },
                      "type": "li",
                    },
                  ],
                  "type": "ul",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SLIDER",
            "className": "LayoutItem",
          },
          "ref": null,
          "rendered": "SLIDER",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Layout"
>
          <div
                    className="LayoutItem LayoutItem--fill LayoutItem--center"
          >
                    <header>
                              <h1
                                        className="Title"
                              >
                                        <span
                                                  aria-label="Wave emoji"
                                                  role="img"
                                        >
                                                  ðŸ‘‹
                                        </span>
                                        I'm Kai
                              </h1>
                    </header>
                    <div
                              className="Content"
                    >
                              <p
                                        className="Quote"
                              >
                                        Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                        <br />
                                        San Francisco, California
                              </p>
                              <ul
                                        className="SocialLinks"
                              >
                                        <li>
                                                  <a
                                                            href="https://www.instagram.com/kaigenji/"
                                                            rel="noopener noreferrer"
                                                            target="_blank"
                                                  >
                                                            @kaigenji
                                                  </a>
                                        </li>
                                        <li>
                                                  <a
                                                            href="mailto:kaigenji@gmail.com"
                                                  >
                                                            email
                                                  </a>
                                        </li>
                              </ul>
                    </div>
          </div>
          <div
                    className="LayoutItem"
          >
                    SLIDER
          </div>
</div>,
        "className": "Page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="LayoutItem LayoutItem--fill LayoutItem--center"
>
              <header>
                            <h1
                                          className="Title"
                            >
                                          <span
                                                        aria-label="Wave emoji"
                                                        role="img"
                                          >
                                                        ðŸ‘‹
                                          </span>
                                          I'm Kai
                            </h1>
              </header>
              <div
                            className="Content"
              >
                            <p
                                          className="Quote"
                            >
                                          Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                          <br />
                                          San Francisco, California
                            </p>
                            <ul
                                          className="SocialLinks"
                            >
                                          <li>
                                                        <a
                                                                      href="https://www.instagram.com/kaigenji/"
                                                                      rel="noopener noreferrer"
                                                                      target="_blank"
                                                        >
                                                                      @kaigenji
                                                        </a>
                                          </li>
                                          <li>
                                                        <a
                                                                      href="mailto:kaigenji@gmail.com"
                                                        >
                                                                      email
                                                        </a>
                                          </li>
                            </ul>
              </div>
</div>,
            <div
              className="LayoutItem"
>
              SLIDER
</div>,
          ],
          "className": "Layout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <header>
                  <h1
                                    className="Title"
                  >
                                    <span
                                                      aria-label="Wave emoji"
                                                      role="img"
                                    >
                                                      ðŸ‘‹
                                    </span>
                                    I'm Kai
                  </h1>
</header>,
                <div
                  className="Content"
>
                  <p
                                    className="Quote"
                  >
                                    Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                                    <br />
                                    San Francisco, California
                  </p>
                  <ul
                                    className="SocialLinks"
                  >
                                    <li>
                                                      <a
                                                                        href="https://www.instagram.com/kaigenji/"
                                                                        rel="noopener noreferrer"
                                                                        target="_blank"
                                                      >
                                                                        @kaigenji
                                                      </a>
                                    </li>
                                    <li>
                                                      <a
                                                                        href="mailto:kaigenji@gmail.com"
                                                      >
                                                                        email
                                                      </a>
                                    </li>
                  </ul>
</div>,
              ],
              "className": "LayoutItem LayoutItem--fill LayoutItem--center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1
                    className="Title"
>
                    <span
                                        aria-label="Wave emoji"
                                        role="img"
                    >
                                        ðŸ‘‹
                    </span>
                    I'm Kai
</h1>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        aria-label="Wave emoji"
                        role="img"
>
                        ðŸ‘‹
</span>,
                      "I'm Kai",
                    ],
                    "className": "Title",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-label": "Wave emoji",
                        "children": "ðŸ‘‹",
                        "role": "img",
                      },
                      "ref": null,
                      "rendered": "ðŸ‘‹",
                      "type": "span",
                    },
                    "I'm Kai",
                  ],
                  "type": "h1",
                },
                "type": "header",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="Quote"
>
                      Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in
                      <br />
                      San Francisco, California
</p>,
                    <ul
                      className="SocialLinks"
>
                      <li>
                                            <a
                                                                  href="https://www.instagram.com/kaigenji/"
                                                                  rel="noopener noreferrer"
                                                                  target="_blank"
                                            >
                                                                  @kaigenji
                                            </a>
                      </li>
                      <li>
                                            <a
                                                                  href="mailto:kaigenji@gmail.com"
                                            >
                                                                  email
                                            </a>
                      </li>
</ul>,
                  ],
                  "className": "Content",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in",
                        <br />,
                        "San Francisco, California",
                      ],
                      "className": "Quote",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Iâ€™m a designer and photographer born in Oakland, raised in Berkeley, commenced in San JosÃ©, and living in",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                      "San Francisco, California",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <li>
                          <a
                                                    href="https://www.instagram.com/kaigenji/"
                                                    rel="noopener noreferrer"
                                                    target="_blank"
                          >
                                                    @kaigenji
                          </a>
</li>,
                        <li>
                          <a
                                                    href="mailto:kaigenji@gmail.com"
                          >
                                                    email
                          </a>
</li>,
                      ],
                      "className": "SocialLinks",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <a
                            href="https://www.instagram.com/kaigenji/"
                            rel="noopener noreferrer"
                            target="_blank"
>
                            @kaigenji
</a>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "@kaigenji",
                            "href": "https://www.instagram.com/kaigenji/",
                            "rel": "noopener noreferrer",
                            "target": "_blank",
                          },
                          "ref": null,
                          "rendered": "@kaigenji",
                          "type": "a",
                        },
                        "type": "li",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <a
                            href="mailto:kaigenji@gmail.com"
>
                            email
</a>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "email",
                            "href": "mailto:kaigenji@gmail.com",
                          },
                          "ref": null,
                          "rendered": "email",
                          "type": "a",
                        },
                        "type": "li",
                      },
                    ],
                    "type": "ul",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SLIDER",
              "className": "LayoutItem",
            },
            "ref": null,
            "rendered": "SLIDER",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
